File: Data Structures & Algorithms/k-closest-points-to-origin/submission-5.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        heap = []


        for a, b in points:
            dist = a*a + b*b
            heap.append([dist, (a, b)])


        heapq.heapify(heap)
        res = []
        for i in range(k):
            a = heapq.heappop(heap)[1]
            res.append(a)

        return res

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-5.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> DP(nums.size(), 1);
        int res =1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] < nums[j]) {
                    DP[i] = max(DP[i], 1 + DP[j]);
                    res = max(res, DP[i]);
                }
            }
        }
        return res;
    }
};
